### DISCUSSION TITLE
Can't pair to dexcom G7

### DISCUSSION THREAD

**Discussion by richw42** [NightscoutFoundation/xDrip#3638] [Bug reports] [answered]
After I paired with my most recent g7, I got only one lonely data point in the next several hours. So I went through the procedure to try reconnecting with the g7, and now I don't even get a pair request. 
Phone is Motorola one 5G ace
Android 11
Edit: I successfully read the Dexcom on my Tandem pump. I don't have any other Dexcom or Tandem app on my phone.
This problem may be because I didn't edit the transmitter ID for several hours after I changed the sensor.
I uploaded a log dump with a message like "discussion 3638".
An old LG Android (V 7.0) acts much the same way. No pair requests.
The LG Link goes from Connecting for a few minutes, Connected for a couple seconds, and then Disconnected.
I have checked the ID many times. I have gone over the G7 settings several times.
![Screenshot_20240821-135145](https://github.com/user-attachments/assets/35ed7423-764d-45f1-b2ea-27c11b30a1e7)

**Comment by Navid200** [comment]
Have you disabled xDrip battery optimization?
Have you disabled Bluetooth battery optimization?
Are all settings correct?  https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html

**Comment by richw42** [comment]
My tandem pump gets the data very well. xDrip has worked fine in the past (although pairing sometimes took a while). There doesn't seem to be a Dexcom app available for my phone.
I've gone over all the settings instructions I could find, several times.
Although I don't think I ever found a Bluetooth specific battery optimization setting.
I did check the xDrip setting under battery.
The big QR code didn't do anything when I touched it.

**Reply by Navid200** [reply]
I have provided the recommended settings above.  Can you just go over those?  

I have explained where to look for Bluetooth battery optimization here: https://navid200.github.io/xDrip/docs/BluetoothBatteryOpt.html

And the xDrip battery optimization is not an xDrip setting.  It is an Android setting and I have explained it here:  
https://navid200.github.io/xDrip/docs/BatteryOpt.html

**Reply by richw42** [reply]
Yes, I finally found the Bluetooth setting, it is not optimized.
And I have gone over all the settings mentioned above.
I have just sent my log to Jamorham.

**Reply by Navid200** [reply]
Do you have t:connect installed on the same phone and it is paired?

**Reply by richw42** [reply]
I don't have t:connect on the phone.

**Comment by richw42** [comment]
Could an Android update have broken things?
My last update was Aug. 11, which would have been after my previous G7 was started (Aug. 9)

**Reply by Navid200** [reply]
An Android update could have reset battery optimizations back to default, which you have verified are OK.

Can you go to the list of connected devices on your phone and delete all starting with DXCM or DX02?
Then, try this to see if you get a pair request:  https://navid200.github.io/xDrip/docs/MissedPairRequest.html

**Comment by richw42** [comment]
All the DX devices are long gone. In fact, I was paired to this sensor for
quite a while and was not getting any readings so I forgot it and have not
been able to connect since. I followed your directions, and the attached
picture was more or less what I saw for more than 25 minutes. No pair
requests.

**Reply by Navid200** [reply]
There is no image attached.  If you log into GitHub instead of using your email client, you will be able to easily attach an image and you will also be able to see exactly what everyone else will see.

Please don't reply to emails for posting. 
It is a fault of GitHub that states in email that you can respond by replying to email.

**Reply by richw42** [reply]
(I don't see what others see; after I attach the screenshot, I just see a bunch of text)

**Comment by richw42** [comment]
Screenshot 
![Screenshot_20240821-171203](https://github.com/user-attachments/assets/4ca01f5e-a865-45a7-a793-f473e235d4ba)

**Reply by Navid200** [reply]
What I don't have is experience with a pump.  Do I understand correctly that the pump itself directly connects to G7?

**Reply by richw42** [reply]
Yes, I use the pump to stop the old sensor, start the new sensor, and enter the code twice. It usually connects immediately (unlike xDrip, which usually takes about 10 to 15 minutes). And occasionally I use the pump to calibrate.
One thing that was different (but shouldn't matter?) is that when I installed the new sensor, I didn't give xDrip the new sensor code for many hours. I just ignored xDrip.

**Reply by Navid200** [reply]
Is there a safe way to disconnect the pump from G7 for 20 minutes?
I don't  want you to compromise your control.  So, you should not do anything.  I am just asking questions.

This is something I have seen before and have  communicated my concerns with the developers.  We will hopefully have a solution soon.

This is what I see:
G7 is not looking for a new buddy!  It seems that G7 is connected to the pump and is not looking for connectivity on the mobile  slot.  This should not happen because the pump and xDrip use different slots.

G7 goes into scan mode after 15 minutes of no connectivity.  I expected this to happen even if a pump was connected.  But, it does not seem to happen.  The fact that you are not getting a pair request shows that the transmitter is not even accepting any requests for connection.

My thought is that if and only if there was a way to disconnect the pump from G7 without compromising your control, G7 will go into scan mode after 15 minutes.  At that point, xDrip should ask for a pair request.

Another problem I have noticed and hope that we can improve is that the connection right after a pair approval is not guaranteed.  I personally usually tap on restart collector 4 minutes after pair approval to guarantee that a second consecutive connection will be made.
The fact that transmitter days on status page shows 12 days means that  there was never a consecutive connection to your transmitter.

After xDrip makes a consecutive connection, I hope that if you then reenable the pump, it will connect.  And if at that point, the pump fails to connect, you will have to disable collection on xDrip to restore the pump and forget xDrip until we figure out what's going on.

PLEASE do not do anything unless there is a safe way to temporarily disconnect the pump.
Even then, if you choose not to, please don't think twice about it and please do NOT do it.

**Reply by richw42** [reply]
I tried putting the pump a long ways away for a half hour, and saw no improvement. 
I hope that when I go to my next sensor, things will be back to normal.

**Reply by Navid200** [reply]
You are sure that 3228 is the pairing code of your current G7?

**Reply by richw42** [reply]
I've checked it about 23 times :)
Now 24.

**Reply by Navid200** [reply]
Do you have another phone in the house to test?  It is possible that this is caused by the phone after all.

**Reply by richw42** [reply]
Well, I tried. LG phone, Android 7.0. Chrome crashes when I try to login to github :( so I'll just have to describe the status page.
Scan Error: Enable Bluetooth and try again
Sensor Status: Mismatch - wait
Trans ID: 3228
Bluetooth Link: Disconnected
Last Connected: 37 minutes ago
Brain State: Scanning Errors: 2

The Last Connected was a multiple of 5 minutes when the data updated on my pump,
so apparently there was some kind of connection once.

I just changed the Trans ID twice. Let's see if anything changes.
BTW, all the settings seemed to be OK (except I couldn't find Bluetooth battery, 
and Allow OB2 unbonding was on and I turned it off)
Now it's Phone Service State: Connect request 13:37 (at 1:38 PM)
Hmmm, the Last Connected time just reset.
Link went from Connecting for a few minutes, Connected for a couple seconds, and now Disconnected.
It did it again, I think 10 minutes later. Now at Power connect Errors: 8
Enough.

**Reply by Navid200** [reply]
What pump?  There is nothing related to a pump in the first post.

If you need help with respect to connectivity, you should clearly identify all the hardware components in the very first post.

**Reply by richw42** [reply]
Sorry, it's the third post or so when we discuss the tandem pump (which on the first post I didn't believe was relevant).
"My tandem pump gets the data very well." And I answered that I didn't have t:connect on the phone. And my phone won't load the Dexcom app.
Oh, and here's a screenshot on my old LG phone
![Screenshot_2024-08-23-14-10-00](https://github.com/user-attachments/assets/80b705ef-e0b2-4017-883a-772121be3bb5)
e:

**Reply by Navid200** [reply]
You cannot expect our lead developer to read every single post here.
Edit the first post and include everything that he needs to know.  The pump is extremely important.

And you need to update xDrip to the latest Nightly as well when you have connectivity problems.

**Comment by jamorham** [comment]
There are two likely causes for "mismatch" error message:

1) The pairing code is not right - double check it
2) There are other old sensors laying nearby which are getting connected to. Move any old sensors far away so they cannot interfere with the pairing process.

**Reply by richw42** [reply]
I don't believe either is the case. I have checked the pairing code about 56 times. And I put the old sensors in an Altoids tin. (although I have removed them now, and they are in the garbage. Could they still be active days after being removed?) The only other live Bluetooth device around is my Panasonic DECT phone, which my cell is paired with.
Perhaps the sensor just refuses to pair some time after it's started.
Mismatch messages were common in the past (which is why I started using the Altoids tin), and it would take maybe 10 or 15 minutes to connect to the new sensor. The big difference this time is that I delayed entering the new pairing code in xDrip. 
Another tidbit: Apparently the new sensor did pair at first, but I got only one lonely reading. So I "forgot" it in Bluetooth settings, and it has never re-paired.
New sensor started Aug. 19, 12:31 PM. One lonely data point, Aug. 20 6:57 AM.

**Reply by Navid200** [reply]
G7 devices remain active about 100 days:  
https://navid200.github.io/xDrip/docs/Battery-condition.html

**Reply by richw42** [reply]
OK, I took out the garbage and I now get fewer periodic connections. But I still see Mismatch. Maybe the garbage container isn't far enough away?
How do you measure the G7 voltage for more than 10.5 days?

**Reply by Navid200** [reply]
A G7 only stops providing readings after 10.5 days.
But, it does not stop transmitting.  You can continue to establish connectivity to it the same as a G6 transmitter.  When a G6 transmitter stops providing readings after 110 days, it does not stop transmitting.

**Reply by Navid200** [reply]
You have to disable collection on a phone to connect with another: https://navid200.github.io/xDrip/docs/Stop-xDrip.html  

Have you restarted the phone?

**Reply by richw42** [reply]
(I've restarted the phone a number of times)
I'm giving up. I'll be back if my next sensor doesn't work :)

**Comment by richw42** [comment, marked as answer]
My next G7 sensor paired in five minutes. So I think the problem was that I didn't connect to the previous sensor immediately after I inserted it. Apparently it refuses to pair after a while.
